src/ui/server/controllers/insurance/business/map-and-save/company-details/index.ts

```diff
@@ -14,20 +17,51 @@ import mapSubmittedData from '../../map-submitted-data/company-details';
 const companyDetails = async (formBody: RequestBody, application: Application, validationErrors?: ValidationErrors) => {
   try {
     if (hasFormData(formBody)) {
-      let saveResponse;
+      console.info('Mapping and saving application - business - company details');
 
       const populatedData = mapSubmittedData(formBody);
 
+      let saveResponse;
+
+      /**
+       * If validation errors, save the data with only valid data.
+       * Otherwise, simply save all data.
+       */
+
       if (validationErrors) {
-        saveResponse = await save.companyDetails(application, populatedData, validationErrors.errorList);
+        saveResponse = await saveCompany.companyDetails(application, populatedData, validationErrors.errorList);
       } else {
-        saveResponse = await save.companyDetails(application, populatedData);
+        saveResponse = await saveCompany.companyDetails(application, populatedData);
       }
 
       if (!saveResponse) {
         return false;
       }
 
+      /**
+       * If DIFFERENT_TRADING_ADDRESS data should be nullified,
+       * Nullify and save the data.
+       */
+      const { hasDifferentTradingAddress } = formBody;
+
+      const {
+        company: {
+          differentTradingAddress: { fullAddress },
+        },
+      } = application;
+
+      if (shouldNullifyCompanyDifferentAddress(hasDifferentTradingAddress, fullAddress)) {
```

Comment:
should we put this in a helper and return the saveResponse from that helper?

Reply:
Good shout, moved into a new function.